File: t/VB/OmrProcessingDemo/Operations/LoadWorkspaceOperation.vb
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Runtime.Serialization.Formatters.Binary
Imports System.Text
Imports System.Threading.Tasks

Class LoadWorkspaceOperation
   Inherits BusyOperation

   Private filename As String
   Private myworkspace As Workspace

   Public ReadOnly Property LoadedWorkspace As Workspace
      Get
         Return myworkspace
      End Get
   End Property

   Public Sub New(ByVal filename As String)
      Me.filename = filename
   End Sub

   Protected Overrides Sub Run()
      Progress(101, "Loading workspace file...")
      Dim bf As BinaryFormatter = New BinaryFormatter()

      Using fs As FileStream = File.OpenRead(filename)
         myworkspace = CType(bf.Deserialize(fs), Workspace)
      End Using

      Progress(101, "Unpacking...")
      myworkspace.Unpack()
      MyBase.Run()
   End Sub
End Class

File: t/CS/OmrProcessingDemo/Operations/LoadWorkspaceOperation.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Runtime.Serialization.Formatters.Binary;
using System.Text;
using System.Threading.Tasks;

namespace OmrProcessingDemo.Operations
{
   class LoadWorkspaceOperation : BusyOperation
   {
      private string filename;
      private Workspace workspace;

      public Workspace LoadedWorkspace { get { return workspace; } }

      public LoadWorkspaceOperation(string filename)
      {
         this.filename = filename;
      }

      protected override void Run()
      {
         Progress(101, "Loading workspace file...");

         BinaryFormatter bf = new BinaryFormatter();

         using (FileStream fs = File.OpenRead(filename))
         {
            workspace = (Workspace)bf.Deserialize(fs);
         }

         Progress(101, "Unpacking...");
         workspace.Unpack();

         base.Run();
      }
   }
}
